patches/chromium/add_ui_scopedcliboardwriter_writeunsaferawdata.patch

```diff
@@ -0,0 +1,60 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Henri Torgemane <henrit@gmail.com>
+Date: Thu, 23 Sep 2021 21:30:33 -0500
+Subject: add ui::ScopedCliboardWriter::WriteUnsafeRawData
+
+This restores some ability to write to the clipboard using raw formats, which
+was removed as part of the Raw Clipboard API scrubbing.
+https://bugs.chromium.org/p/chromium/issues/detail?id=1217643
+
+diff --git a/ui/base/clipboard/clipboard_metrics.h b/ui/base/clipboard/clipboard_metrics.h
+index 044655a9be7117e13842adfbf07db5da1b85e234..47c743fac86bccb4ed7aefb134d8bb7e7c39ac04 100644
+--- a/ui/base/clipboard/clipboard_metrics.h
++++ b/ui/base/clipboard/clipboard_metrics.h
+@@ -22,7 +22,8 @@ enum class ClipboardFormatMetric {
+   kSvg = 8,
+   kFilenames = 9,
+   kPng = 10,
+-  kMaxValue = kPng,
++  kUnsafeRawData = 11,
++  kMaxValue = kUnsafeRawData,
+ };
+ 
+ void RecordRead(ClipboardFormatMetric metric);
+diff --git a/ui/base/clipboard/scoped_clipboard_writer.cc b/ui/base/clipboard/scoped_clipboard_writer.cc
+index 153f169d2cdef6f8a726c188283a5bc1b7395fa3..4c0f39f88b9c7c33e9b1b5d0654d3612d0d961f7 100644
+--- a/ui/base/clipboard/scoped_clipboard_writer.cc
++++ b/ui/base/clipboard/scoped_clipboard_writer.cc
+@@ -212,6 +212,17 @@ void ScopedClipboardWriter::WriteData(const std::u16string& format,
+   }
+ }
+ 
++void ScopedClipboardWriter::WriteUnsafeRawData(const std::u16string& format,
++                                               mojo_base::BigBuffer data) {
++  RecordWrite(ClipboardFormatMetric::kUnsafeRawData);
```

Comment:
Chromium only uses this to record perf metrics - we can remove this and the added enum class value as we don't make use of them in Electron

Reply:
Thanks for the quick feedback. That makes sense. 
What would be a good folder to put the subclass? 

Should it go next to the rest of the code in `shell/common/api/`, or maybe one-level up in `shell/common/` since it doesn't itself define apis?

Reply:
let's just go with `shell/common/` for now!

Reply:
@henrit I looked into this and I agree with you on your assessment.  Given that we would have to bring over a large amount of the the clipboard code, we should instead proceed with the patch instead.  @henrit are you willing to continue working on this?  If not, I'd be happy to take it over, but since it's your PR, I'd prefer to defer to you.

Looking at the PR, I think if it is rebased to resolve the conflict and perhaps cleanup the patch with @codebytere's suggestions, we can move this PR forward.

cc @ckerr @codebytere

Reply:
I'm a little tardy with this response, but FWIW I've looked into this wrt pickling and also agree with trying a patch first due to the amount of code that we'd need to copy over if we copied vs patching.

Some of my review comments don't make sense in a patch context. Feel free to mark resolved at will.

I'm still happy to help via followup review and/or pairing if anyone wants it, whether it's @henrit or @jkleinsc.